DESCRIPTION

```diff
@@ -36,16 +36,19 @@ URL: https://insightsengineering.github.io/teal/,
 BugReports: https://github.com/insightsengineering/teal/issues
 Depends:
     R (>= 4.0),
-    shiny (>= 1.7.0),
+    shiny (>= 1.8.1),
     teal.data (>= 0.5.0),
     teal.slice (>= 0.5.0)
 Imports:
     checkmate (>= 2.1.0),
+    future,
     jsonlite,
     lifecycle (>= 0.2.0),
     logger (>= 0.2.0),
     magrittr (>= 1.5),
     methods,
+    promises,
+    renv,
```

Comment:
can you please specify min version supported? I would like to avoid `verdepcheck` failures after the merge. Feel free to execute this manually for that feature branch.

The current setting indicates compatibility with *all* historical versions of these packages and for instance [`future@0.6.0` does not have / not export `multisession`](https://github.com/cran/future/blob/0.6.0/NAMESPACE) object you are using.

Reply:
Sure @pawelru - very good point

Reply:
on it

Reply:
@averissimo starter a verdepcheck CI build in here - will investigate what are the results https://github.com/insightsengineering/teal/actions/runs/9568766694/job/26406371797

Reply:
| package | max | min_cohort | min_isolated | release |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| future | 1.33.2-9007 | 1.33.2 | x | fail |
| promises | 1.2.1.9001 | 1.3.0 | x | fail |
| renv | 1.0.7.900 | 1.0.7 | x | fail |
  
x - still building  
  
release failed due to 

```r
Can't install dependency teal.reporter (>= 0.3.1.9004)
```

Reply:
release strategy is failing because there is no released version of `teal.reporter` that would meet version criterion. This is an expected failure. We are more interested in the min strategies

The table below shows packages and its versions _used_ in the test run. These are derived automatically following the some logic specific to the strategy used. This does not tell you the correct version! It might happen that the R CMD CHECK of `teal` with super low version of dependencies will fail and in such case you have to increase it. 

Note that if you don't specify the version (the current state) then the lowest possible is used and it's quite likely that we will fail. (unless the code is fully compatible with the packages from long long time ago).

Reply:
min_isolated froze for vignettes build https://github.com/insightsengineering/teal/actions/runs/9568766694/job/26406372023#step:9:418 for more than 2 hours so I guess it will never get build.

Is it fine to assume that we can put versions from `min_cohort`?

Reply:
Yes. The expected time is <10 mins I would say. Please stop the job. I don't know what's going on there.

Please try also locally. Maybe that would help.

Reply:
I did stop the job. Locally when I run on Windows I get

```r
✔ Downloaded 135 packages (152.37 MB) in 5m 8.8s                                                
Error in rawToChar(block[seq_len(ns)]) : 
  embedded nul in string: 'PK\003\004\n\0\0\0\0\0\xb4C\xa2X\0\0\0\0\0\0\0\0\0\0\0\0\b\0\0\0Biobase/PK\003\004\024\0\002\0\b\0\xa8C\xa2X\032\xdc^\x89\xde\001\0\06\003\0\0\020\0\0\0Biobase/CITATION\x9dR\xc1n\xdb0\f\xbd\xe7+\x88\\\xd6\002\x8e<'
```

for all 4 strategies. It looks like we are not there with Windows support.


I will try to specify versions from `min_isolated` and let's see if R CMD check succeeds on this PR